app/controllers/api/v1/application_controller.rb

```diff
@@ -11,19 +13,14 @@ def current_user
         User.find(doorkeeper_token.resource_owner_id) if doorkeeper_token
       end
 
-      # Render Error Message in json_api format
-      # :reek:LongParameterList { max_params: 5 }
-      def render_error(detail:, source: nil, meta: nil, status: :unprocessable_entity, code: nil)
-        errors = [
-          {
-            source: source,
-            detail: detail,
-            code: code,
-            meta: meta
-          }.compact
-        ]
+      def doorkeeper_unauthorized_render_options(error: nil)
+        return unless error
 
-        render json: { errors: errors }, status: status
+        {
+          json: {
+            errors: build_errors(detail: error.description, source: error.state, code: error.name)
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/nimblehq/nimble-survey-web/blob/700447c0a197333d7c81d82b62cd48614d8bc4d0/app/controllers/api/v1/application_controller.rb#L21"></span>Refers to ‘error’ more than self (maybe move it to another class?)</td>
    </tr>
  </tbody>
</table>